crates/pathfinder/src/state/sync.rs

```diff
@@ -48,15 +48,15 @@ pub async fn sync<Transport, SequencerClient, F1, F2, L1Sync, L2Sync>(
     chain: Chain,
     sequencer: SequencerClient,
     state: Arc<State>,
-    l1_sync: L1Sync,
+    mut l1_sync: L1Sync,
     l2_sync: L2Sync,
 ) -> anyhow::Result<()>
 where
     Transport: EthereumTransport + Clone,
     SequencerClient: sequencer::ClientApi + Clone + Send + Sync + 'static,
     F1: Future<Output = anyhow::Result<()>> + Send + 'static,
     F2: Future<Output = anyhow::Result<()>> + Send + 'static,
-    L1Sync: FnOnce(mpsc::Sender<l1::Event>, Transport, Chain, Option<StateUpdateLog>) -> F1 + Copy,
+    L1Sync: FnMut(mpsc::Sender<l1::Event>, Transport, Chain, Option<StateUpdateLog>) -> F1,
     L2Sync: FnOnce(
```

Comment:
Should `L2Sync` change to match?

Reply:
I actually thought about this but didn't see any other affected test, or any other test to rewrite so didn't want to go through this, yet at least.